### DISCUSSION TITLE
How Input Settings Works ?

### DISCUSSION THREAD

**Discussion by raphael2228** [Adriwin06/Ultimate-CommonUI-Menu-System#12] [Q&A] [answered]
Hi, can you explain how the input mapping system works? I'm trying to implement it on my own project. But there's nothing displayed in game and when I quit I get this error  "Blueprint Runtime Error: "Accessed None trying to read property CallFunc_GetUserSettings_ReturnValue". Node:  Branch Graph:  CreateInputMapping Function:  Create Input Mapping Blueprint:  BFL_UI"

**Comment by Adriwin06** [comment]
Hi! Can you share some screenshots please? Is it this branch on this function that cause this issue?
![image](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/76881633/8a036686-e6a9-4092-b884-aa8934ff1534)

**Reply by raphael2228** [reply]
YES, it's at the branch that things get stuck. the input action works correctly otherwise.

**Reply by raphael2228** [reply]
![Capture d’écran 2024-06-02 214711](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/1048eac6-9407-4ea0-850a-df138e579289)
![Capture d’écran 2024-06-02 214638](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/55704a1b-400d-4d02-8c1a-3df24bb3078e)
![Capture d’écran 2024-06-02 214621](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/5e7aef22-c2be-4760-b0c3-097c25852c25)
i purposely didn't put the “apply settings” since i already have mine. but it doesn't affect the inputs so i don't think it's that.

**Reply by Adriwin06** [reply]
When you said that nothing was displayed in-game, were you talking about the key binding menu? Do you use your own Input Action Mapping? If so, you probably forgot to set your Input Action to use Player Mappable Key. If not, I will investigate further because I have never encountered this issue.

**Reply by raphael2228** [reply]
I meant that nothing is displayed in the settings menu when you change the input . I don't use your system directly, but I'm trying to adapt it to mine, so as you say, I must have forgotten something, but I don't know what.

**Reply by Adriwin06** [reply]
Are you sure you added the Player Mappable Key Settings ?

![image](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/76881633/a7ce2de8-9fea-417e-ab80-f314d9316bc1)

**Reply by raphael2228** [reply]
![Capture d’écran 2024-06-04 125603](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/e07e5376-592f-4d49-b4f6-3d429ecd35f0)
![Capture d’écran 2024-06-04 125628](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/bd3be8c3-09ff-4cdb-b2fc-809aafea0c42)
![Capture d’écran 2024-06-04 125701](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/1719e45f-2dea-4858-bafc-6db38fb3484c)

**Reply by Adriwin06** [reply]
Sorry I really don't see where the problem is. If it's on your end I'm not sure I can really help.

**Reply by raphael2228** [reply, marked as answer]
> Sorry I really don't see where the problem is. If it's on your end I'm not sure I can really help.

I finally found the problem. I didn't have the associated option checked. in new projects it's probably checked by default. but not if it comes from unreal engine 4 initially.
![ue](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/63161462/581947fd-fa3d-463a-b225-5b8fb4bbc39e)

**Reply by Adriwin06** [reply]
Oh, I should have thought of that. Sorry. But you managed to find the issue and fix it, so it's all good. Let me know if you have any other questions, and I hope your experience with my project will be great 👍

**Comment by raphael2228** [comment]
after check, the get enhanced input user settings returns nothing if I test it with a “print string”, unlike your demo.

**Reply by Adriwin06** [reply]
Are you sure you have these settings in Project Settings>Input?

![image](https://github.com/Adriwin06/Ultimate-CommonUI-Menu-System/assets/76881633/c3b46d00-7680-40e9-a739-5e77599d50e2)

**Reply by raphael2228** [reply]
yes, the buttons work without a problem. i can move, open a menu. that's why i don't understand.
